### ISSUE TITLE
[V3] Implement new Concept Exercise: types

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1378]
WARNING: this exercise probably be eliminated. Don't implement.

This issue describes how to implement the `types` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to give the student a rough overview of Go's types.

## Learning objectives

- overview of basic types (string, bool, numbers)
- overview of composite types: see [type-system-overview](https://go101.org/article/type-system-overview.html)
- declare custom type
- declare a variable of a type T
- type conversion

## Out of scope

- working with a type beyond declaring and returning it
- filling a composite type with values

## Concepts

- `types`
- `ints`
- `booleans`
- `strings`
- `maps`
- `slices`
- `functions`
- `channels`
- `pointers`

## Prerequisites

None!

## Resources to refer to

- https://go101.org/article/type-system-overview.html
- https://golang.org/ref/spec#Types

## Implementing

Some ideas of functions to implement. Note that the the function definitions themselves should be part of the stub provided to the student.

// Edit: worked on it a bit more and created a little story...

```go
// BakeFruitCake executes the steps of baking a cake.
// This is already implemented, but how to perform the individual steps is still a mystery.
func BakeFruitCake() {
    GreetChef()

    // To preheat the oven:
    PreheatOven(GetDegrees())

    // To check how much flour is available:
    CheckFlour()

    // Get fruit list for the cake (it's a fruit cake!):
    GetFruits()

    // ... and that list of weights -- where was it again?
    fruitWeights := GetAvgWeights()

    // How many apples do we need again? Let's create an apple weight incrementer:
    appleWeightIncr := NewIncrementer(weights["apples"])

    // incrementing: (todo: replace with loop... where is that loop recipe??)
    appleWeight := appleWeightIncr(0)
    appleWeight = appleWeigthIncr(appleWeight)
    appleWeight = appleWeigthIncr(appleWeight)

    // the story goes on, but we don't want to bore you with any more details...
}

func GreetChef() string {
    // To be written by the student
    return "Hello Chef!"
}

func GetDegrees() int {
    // To be written by the student
    return 180
}

// CheckFlower returns the amount of available flower in gram.
// Note: sometimes the information is not available.
// In order to differentiate between 
// - 0 flower available
// - no available information
// the return value is nullable.
func CheckFlower() *int {
    // To be written by the student
   n := 15473
   return &n
}

// To be defined by the student
type weights map[string]int

// GetAvgWeights returns a list of average weights of fruits in gram.
func GetAvgWeights() weights {
    // To be written by the student
    return weights{
        "apples": 182,
        "bananas": 115,
        "oranges": 169,
        "peaches": 128,
    }
}

// Fruits returns a list of fruit names.
func Fruits() []string {
    // To be written by the student
    return []string{
        "apples",
        "bananas",
        "oranges",
        "peaches",
    }
}

func NewIncrementer(incr int) func(base int) int {
    // To be written by the student
    return func(base int) int {
        // some basic intro on adding numbers needed?
        return base + incr
    }
}

// PreheatOven converts the degrees celcius in the only unit the oven understands.
// Naturally that is uint32. What else could it possibly be?
func PreheatOven(n int) uint32 {
    // To be written by the student
    return uint32(n)
}
```

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
